strawberry_django/routers.py

```diff
@@ -0,0 +1,68 @@
+from __future__ import annotations
```

Comment:
I looked at the original too, and that's the one that put me on the wrong trail related to authentication.  
As I was trying to make things more 'battery included' it seems like a wise idea to include the new version.  

Since the original doesn't leave much room for subclassing I decided to copy-paste it, and add the authentication to avoid breaking other peoples installs.